src/ray/gcs/redis_client.cc

```diff
@@ -200,6 +200,13 @@ Status RedisClient::Connect(std::vector<instrumented_io_context *> io_services)
   return Status::OK();
 }
 
+void RedisClient::ReattachContext(RedisContext &context) {
+  // TODO ideally we want to remove the old one too
```

Comment:
why don't we remove?

Reply:
The main reason was to keep the implementation simple. I guess we can modify the deleter of the old context to remove this as well. (When a new version of the context is added, there might still be live references to the old context, so we can't remove it yet.)